### PR TITLE
EOS-19798: updates in mismatch param names

### PR DESCRIPTION
**Pull request by divya-vijay**
Signed-off-by: Divya Vijayakumar <divya.vijaykumar@seagate.com>

### CODE REVIEW

File: api/python/provisioner/inputs.py [hunk 1]
```diff
@@ -819,9 +822,6 @@ class NetworkParams():
     primary_data_netmask: str = ParamGroupInputBase._attr_ib(
         _param_group, descr="primary node data interface netmask"
     )
-    primary_data_network_iface: List = ParamGroupInputBase._attr_ib(
```
Comment (divya-vijay) [inline api/python/provisioner/inputs.py:822]: Remove `iface` in the hope that we're no longer using it and it was not found in `params_spec` as well. Do we need to include it?

Comment (ypise) [reply api/python/provisioner/inputs.py:822]: It should be renamed to data_public_interfaces and data_private_interfaces,
and related changes to be accompanied

File: api/python/provisioner/inputs.py [hunk 2]
```diff
@@ -859,9 +862,6 @@ class NetworkParams():
     secondary_data_netmask: str = ParamGroupInputBase._attr_ib(
         _param_group, descr="secondary node data interface netmask"
     )
-    secondary_data_network_iface: List = ParamGroupInputBase._attr_ib(
```
Comment (ypise) [inline api/python/provisioner/inputs.py:862]: Secondary can be deleted as each node has individual identity.
Plus the number of nodes are uncertain and should be queried from pillar data.

File: api/python/provisioner/inputs.py [hunk 3]
```diff
@@ -883,24 +883,24 @@ class Network(ParamGroupInputBase):
     search_domains: List = NetworkParams.search_domains
     primary_hostname: str = NetworkParams.primary_hostname
     primary_data_roaming_ip: str = NetworkParams.primary_data_roaming_ip
+    primary_data_floating_ip: str = NetworkParams.primary_data_floating_ip
     primary_mgmt_public_ip: str = NetworkParams.primary_mgmt_public_ip
     primary_mgmt_netmask: str = NetworkParams.primary_mgmt_netmask
     primary_mgmt_gateway: str = NetworkParams.primary_mgmt_gateway
     primary_data_netmask: str = NetworkParams.primary_data_netmask
     primary_data_gateway: str = NetworkParams.primary_data_gateway
-    primary_data_network_iface: List = NetworkParams.primary_data_network_iface
```
Comment (ypise) [inline api/python/provisioner/inputs.py:891]: Need this.
Drop primary from keys and all secondary keys.

File: api/python/provisioner/inputs.py [hunk 4]
```diff
@@ -883,24 +883,24 @@ class Network(ParamGroupInputBase):
...
     primary_data_public_ip: str = NetworkParams.primary_data_public_ip
     primary_bmc_ip: str = NetworkParams.primary_bmc_ip
     primary_bmc_user: str = NetworkParams.primary_bmc_user
     primary_bmc_secret: str = NetworkParams.primary_bmc_secret
     secondary_hostname: str = NetworkParams.secondary_hostname
-    secondary_floating_ip: str = NetworkParams.secondary_floating_ip
+    secondary_data_roaming_ip: str = NetworkParams.secondary_data_roaming_ip
+    secondary_data_floating_ip: str = NetworkParams.secondary_data_floating_ip
```
Comment (ypise) [inline api/python/provisioner/inputs.py:898]: Floating is not a requirement yet and should be dropped for now.

File: api/python/provisioner/params_spec.yaml [hunk 5]
```diff
@@ -29,7 +29,8 @@ network:
   #'key_path': ('cluster', 'srvnode-2', 'network', 'data', 'public_ip'),
   # TODO use nested group to reduce copy-paste in paths
   primary_hostname: cluster/srvnode-1/hostname
-  primary_roaming_ip: cluster/srvnode-1/network/data/roaming_ip
+  primary_data_roaming_ip: cluster/srvnode-1/network/data/roaming_ip
+  primary_data_floating_ip: cluster/srvnode-1/network/data/floating_ip
```
Comment (ypise) [inline api/python/provisioner/params_spec.yaml:33]: floating should be dropped.

File: api/python/provisioner/params_spec.yaml [hunk 6]
```diff
@@ -41,7 +42,8 @@ network:
   primary_bmc_user: cluster/srvnode-1/bmc/user
   primary_bmc_secret: cluster/srvnode-1/bmc/secret
   secondary_hostname: cluster/srvnode-2/hostname
-  secondary_roaming_ip: cluster/srvnode-2/network/data/roaming_ip
+  secondary_data_roaming_ip: cluster/srvnode-2/network/data/roaming_ip
+  secondary_data_floating_ip: cluster/srvnode-2/network/data/floating_ip
```
Comment (ypise) [inline api/python/provisioner/params_spec.yaml:46]: secondary keys should be dropped.

**Review by ypise — CHANGES_REQUESTED** [review]

**Review by ypise — APPROVED** [review]

**Pull request merged**